### DISCUSSION TITLE
Help me understand binding...

### DISCUSSION THREAD

**Discussion by jamesonuk** [Koenkk/zigbee2mqtt#17336] [Devices]
So reading https://www.zigbee2mqtt.io/guide/usage/binding.html makes me think this would be really useful to implement so I can have my Zigbee switches directly control the lights rather than relying on an automation in HA.

I have some Hue in wall switches (https://www.zigbee2mqtt.io/devices/929003017102.html) and Ikea bulbs in groups.

After some messing around I have been able to bind the switch and group but it doesn't do anything..

Would I be right in thinking that even though a device has a client (output) cluster for On/Off and the lights have a server (input) cluster this is not enough for the switch to control the lights?   (Or I am doing something wrong?)

What I am struggling with is what the clusters actually mean.   My understanding is that by supporting a cluster the device is saying it complies with the Zigbee standard for that cluster so messages standard to / from it should work?    The fact the above doesn't work makes me think that isn't the case in which case I can't see why it has the On/Off cluster if it doesn't support it?

Similarly I have some radiator valves which when used with the supplied controller allow you to set temperature via an external thermostat (ie. the thermostat overrides the value on the trv itself).   I can see that this thermostat has a client hvacThermostat cluster and the trv itself has a server hvacThermostat cluster but the docs say

> Possible clusters are: genScenes, genOnOff, genLevelCtrl, lightingColorCtrl and closuresWindowCovering

which suggests hvacThermostat isn't something the can be bound?   Is that a limitation of Z2M or Zigbee (just that I believe the native controller uses binding to achive this).

Can anyone explain the lower level mechanics behind this?

**Comment by helgek** [comment]
I have the same questions, would be great to benefit from some knowledge transfer.

**Comment by ghost** [comment]
I tried with a Hue remote controller and Ikea bulbs in a group. It seems I had to bind the genBasic cluster to be able to turn on/off the bulbs, otherwise I was only able to control the brightness.

Unfortunately I can't help much more as I've ditched the Hue remote for now, so I can't double check that :|

Hoping that it can help though...
